Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2346 -- Internal Docs in Reader Sidebar

Resolves [EREGCSC-2346](https://jiraent.cms.gov/browse/EREGCSC-2346)

**Description:**

Adds login link to right sidebar of reader view.  Once logged in, users will see internal documents related to selected subpart or section at the bottom of the right sidebar.

**This pull request changes:**

- `ContentSearchViewset` inherits from correct mixin to allow querying by section 
- Creates login Call To Action template partial for reader view's right sidebar
- Creates `InternalDocsContainer` Vue component to fetch internal categories and docs
- Renders those fetched internal docs using existing `SupplementalContentCategory` Vue component
- Refactors `FormatResourceCategories` utility method to handle either public resources or eternal documents
- Adds some unit test coverage and JSDoc documentation to `FormatResourceCategories` utility method
- Adds `mitt` npm library to project to replace the deprecated Vue 2 event emitter interface in all places that it is used.
   - see official Vue documentation: https://v3-migration.vuejs.org/breaking-changes/events-api.html
   - see recipe used to implement `mitt`: https://binodmahto.medium.com/event-bus-implementation-in-vue-5eb194aed569

**Steps to manually verify this change:**

1. Ensure you are logged out and then visit a subpart or section in the Reader View of the experimental deployment.
   - Ex: [45 CFR 95.1](https://3cdqvh7w57.execute-api.us-east-1.amazonaws.com/dev1107/45/95/Subpart-A/2019-11-05/#95-1)
2. Notice the right sidebar.  
   - There should be a Call to Action banner at the top of the right sidebar inviting the user to log in
   - There should NOT be any Public/Internal labels or internal documents visible at the bottom of the right sidebar
3. Use the Call to Action banner link at the top of the right sidebar to log in.  You should be redirected back to 45 CFR 95.
4. Notice that there are now Public and Internal labels in the right sidebar.  Internal documents should be visible at the bottom of the right sidebar
5. An empty category that is set to "Show if Empty" should be visible and match the disabled state of the empty Public categories
6. An active category with a subcategory should be active and able to be toggled
   - Note that the expand/collapse behavior for internal documents should match the public doc behavior
7. Visit the reader view for any other subpart or section.
   - Ex: [45 CFR 115](https://3cdqvh7w57.execute-api.us-east-1.amazonaws.com/dev1107/45/155/Subpart-A/2023-11-22/)
8. Note that there is only an empty, disabled category shown for Internal docs.  These "Show if Empty" categories will be used as the empty state for subparts and sections that do not have any internal documents attached.
